src/main/java/core/basesyntax/service/file/FileReaderServiceImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.file;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class FileReaderServiceImpl implements FileReaderService {
+
+    @Override
+    public List<String> read(String path) {
+        try {
+            return Files.readAllLines(Path.of(path));
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot read file: " + path, e);
```

Comment:
Consider whether wrapping the `IOException` in a `RuntimeException` is the best approach for your application. Depending on your error handling strategy, you might want to define a custom exception or handle the error in a way that allows for recovery or logging.